File: Sources/CreatorCoreForge/UnifiedVideoEngine.swift
import Foundation

/// Shared video engine wrapper around `GPUVideoRenderer` for cross-app usage.
public final class UnifiedVideoEngine {
    public static let shared = UnifiedVideoEngine()
    private let renderer: GPUVideoRenderer

    public init(renderer: GPUVideoRenderer = GPUVideoRenderer()) {
        self.renderer = renderer
    }

    /// Render frames using the underlying GPU renderer.
    public func render(frames: [String], options: RendererOptions) -> GPUVideoClip {
        renderer.render(frames: frames, options: options)
    }
}

File: VisualLab/src/UnifiedVideoEngine.ts
import {
  GPUVideoRenderer,
  RendererOptions,
  GPUVideoClip,
  RenderedFrame,
} from './GPUVideoRenderer.ts';

export class UnifiedVideoEngine<F = unknown> {
  static shared = new UnifiedVideoEngine();
  private constructor(private renderer = new GPUVideoRenderer<F>()) {}

  async render(
    frames: F[],
    options: RendererOptions
  ): Promise<GPUVideoClip<RenderedFrame<F>>> {
    return this.renderer.render(frames, options);
  }
}
